### ISSUE TITLE
Cocoapods update fail after 1.0.0 update

### ISSUE THREAD

**Issue by danipralea** [CocoaPods/CocoaPods#5278]
### Command

```
/usr/local/bin/pod update
```

### Report

* What did you do?

* What did you expect to happen?

* What happened instead?

### Stack

```
   CocoaPods : 1.0.0
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin15]
    RubyGems : 2.0.3
        Host : Mac OS X 10.11.4 (15E65)
       Xcode : 7.3 (7D175)
         Git : git version 2.6.4 (Apple Git-63)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 35f6b653c2f364c601d2cbddad8d9705b168f3f9
```

### Plugins

```
cocoapods-deintegrate : 1.0.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.0.0
```

### Podfile

```ruby
# Uncomment this line to define a global platform for your project
platform :ios, '8.0'
use_frameworks!

# ignore all warnings from all pods
inhibit_all_warnings!

# Remove 64-bit build architecture from all pod targets and override 'Build Active Architecture Only' to NO.
#post_install do |installer|
#    installer.project.targets.each do |target|
#        target.build_configurations.each do |configuration|
#            target.build_settings(configuration.name)['VALID_ARCHS'] = '$(ARCHS_STANDARD_32_BIT)'
#            target.build_settings(configuration.name)['ONLY_ACTIVE_ARCH'] = 'NO'
#        end
#    end
#end


target '******' do

pod 'Alamofire', '~> 3.0'
#pod 'PBJVision'
pod 'ObjectMapper', '~> 1.1'
#pod 'Instabug'

pod 'SwiftyJSON', :git => 'https://github.com/SwiftyJSON/SwiftyJSON.git'
pod 'JPCActivityIndicatorButton', :git => 'https://github.com/jpchmura/JPCActivityIndicatorButton.git', :branch => 'swift-2.0'
pod 'SwiftValidators'
pod 'AFDateHelper'
pod 'CarbonKit', '~>2.1'


# eyecandy
pod 'PKHUD'
#pod 'ChameleonFramework/Swift'
#pod 'DrawerController', '~> 1.0'
pod 'AlamofireImage', '~> 2.0'
pod 'SnapKit', '~> 0.15.0'
pod 'TSMessages', :git => 'https://github.com/KrauseFx/TSMessages.git'
pod 'MJRefresh'

pod 'RxSwift',    '~> 2.0'
pod 'RxCocoa',    '~> 2.0'
pod 'RxBlocking', '~> 2.0'

pod 'Crashlytics'
pod 'Fabric'

pod 'Google/SignIn'

pod 'GoogleMaps'

pod 'Refresher'
pod 'Infinity'

# pod 'UITags'

pod 'FBSDKCoreKit'
pod 'FBSDKLoginKit'
pod 'FBSDKShareKit'

post_install do |installer|
    installer.pods_project.build_configuration_list.build_configurations.each do |configuration|
        configuration.build_settings['CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES'] = 'YES'
    end
end

end
```

### Error

```
ArgumentError - comparison of Pod::Version with Gem::Version failed
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/sources_manager.rb:133:in `>'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/sources_manager.rb:133:in `verify_compatibility!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/sources_manager.rb:89:in `block (2 levels) in update'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/user_interface.rb:63:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/sources_manager.rb:86:in `block in update'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/sources_manager.rb:85:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/sources_manager.rb:85:in `update'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer/analyzer.rb:220:in `block in update_repositories'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer/analyzer.rb:218:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer/analyzer.rb:218:in `update_repositories'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer.rb:147:in `block in resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/user_interface.rb:63:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer.rb:146:in `resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer.rb:114:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/command/update.rb:60:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-1.0.0/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/command.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

**Comment by arden** [comment]
arden-2:Tujiao arden$ pod update
Update all pods
Updating local specs repositories
Performing a deep fetch of the `master` specs repo to improve future performance

then  Stop here .

**Comment by danipralea** [comment]
that's what I got when I did pod update.
I fixed it by deleting all the cocoapods installations and install the 1.0.0 one

Thanks!

**Comment by mattsellars** [comment]
I'm getting this and I only have cocoa pods 1.0.0 installed

**Comment by iTofu** [comment]
@mattsellars 
I have same question to this man. I install the 1.0.0 only, but show:

````
YJCMBP:LCBlur yuanjincang$ gem list

*** LOCAL GEMS ***

activesupport (4.2.5)
bigdecimal (1.2.0)
CFPropertyList (2.2.8)
claide (1.0.0, 0.9.1)
colored (1.2)
escape (0.0.4)
fourflusher (0.3.0)
fuzzy_match (2.0.4)
i18n (0.7.0)
io-console (0.4.2)
json (1.7.7)
libxml-ruby (2.6.0)
minitest (5.8.3, 4.3.2)
molinillo (0.4.5, 0.4.1)
nap (1.0.0)
netrc (0.7.8)
nokogiri (1.5.6)
psych (2.0.0)
rake (0.9.6)
rdoc (4.0.0)
rubygems-update (2.5.1)
sqlite3 (1.3.7)
test-unit (2.0.0.0)
thread_safe (0.3.5)
tzinfo (1.2.2)
xcodeproj (1.0.0, 0.28.2)
YJCMBP:~ yuanjincang$ sudo gem install cocoapods
Fetching: cocoapods-core-1.0.0.gem (100%)
Successfully installed cocoapods-core-1.0.0
Fetching: cocoapods-deintegrate-1.0.0.gem (100%)
Successfully installed cocoapods-deintegrate-1.0.0
Fetching: cocoapods-downloader-1.0.0.gem (100%)
Successfully installed cocoapods-downloader-1.0.0
Fetching: cocoapods-plugins-1.0.0.gem (100%)
Successfully installed cocoapods-plugins-1.0.0
Fetching: cocoapods-search-1.0.0.gem (100%)
Successfully installed cocoapods-search-1.0.0
Fetching: cocoapods-stats-1.0.0.gem (100%)
Successfully installed cocoapods-stats-1.0.0
Fetching: cocoapods-trunk-1.0.0.gem (100%)
Successfully installed cocoapods-trunk-1.0.0
Fetching: cocoapods-try-1.0.0.gem (100%)
Successfully installed cocoapods-try-1.0.0
Fetching: cocoapods-1.0.0.gem (100%)
Successfully installed cocoapods-1.0.0
Parsing documentation for cocoapods-core-1.0.0
Installing ri documentation for cocoapods-core-1.0.0
Parsing documentation for cocoapods-deintegrate-1.0.0
Installing ri documentation for cocoapods-deintegrate-1.0.0
Parsing documentation for cocoapods-downloader-1.0.0
Installing ri documentation for cocoapods-downloader-1.0.0
Parsing documentation for cocoapods-plugins-1.0.0
Installing ri documentation for cocoapods-plugins-1.0.0
Parsing documentation for cocoapods-search-1.0.0
Installing ri documentation for cocoapods-search-1.0.0
Parsing documentation for cocoapods-stats-1.0.0
Installing ri documentation for cocoapods-stats-1.0.0
Parsing documentation for cocoapods-trunk-1.0.0
Installing ri documentation for cocoapods-trunk-1.0.0
Parsing documentation for cocoapods-try-1.0.0
Installing ri documentation for cocoapods-try-1.0.0
Parsing documentation for cocoapods-1.0.0
Installing ri documentation for cocoapods-1.0.0
9 gems installed
YJCMBP:LCBlur yuanjincang$ pod --version
1.0.0
YJCMBP:LCBlur yuanjincang$ pod setup
Setting up CocoaPods master repo
Performing a deep fetch of the `master` specs repo to improve future performance
````

And then stop here.

**Comment by danipralea** [comment]
This is still occuring. at each update/install

**Comment by mseddon** [comment]
Me too- running pod setup on a fresh install of cocoapods yields:

```
$ pod setup
Setting up CocoaPods master repo
[!] /usr/bin/git clone https://github.com/CocoaPods/Specs.git master

Cloning into 'master'...
error: RPC failed; curl 56 SSLRead() return error -9806
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
```

**Comment by segiddins** [comment]
does running the `git` command manually produce different output?

**Comment by mseddon** [comment]
Nope.
```
$ /usr/bin/git clone https://github.com/CocoaPods/Specs.git master
Cloning into 'master'...
remote: Counting objects: 680376, done.
remote: Compressing objects: 100% (34/34), done.
Receiving objecterror: RPC failed; curl 56 SSLRead() return error -9806
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
```
It hung at 12% for about 15 minutes then died.

Interestingly, this works:
```
$ /usr/bin/git clone git@github.com:CocoaPods/Specs.git master
Cloning into 'master'...
Warning: Permanently added the RSA host key for IP address '192.30.252.131' to the list of known hosts.
Saving password to keychain failed
Identity added: /Users/matt/.ssh/id_rsa ((null))
remote: Counting objects: 680466, done.
remote: Compressing objects: 100% (22/22), done.
remote: Total 680466 (delta 6), reused 0 (delta 0), pack-reused 680442
Receiving objects: 100% (680466/680466), 328.89 MiB | 4.23 MiB/s, done.
Resolving deltas: 100% (283455/283455), done.
Checking connectivity... done.
Checking out files: 100% (86469/86469), done.
```

**Comment by segiddins** [comment]
In that case, I guess it's a git issue not specifically about CocoaPods ?

**Comment by mseddon** [comment]
I don't think CocoaPods itself is to blame here, no.  You could use ssh based pulls but then the user would require a github account.  Perhaps there could be (or already is) a way to specify the protocol to use?

**Comment by segiddins** [comment]
`git config --global url."git@github.com:".insteadOf "https://github.com/"` might be what you're looking for?

**Comment by mseddon** [comment]
Yes! That's it. I had no idea that feature existed, thanks :)

**Comment by MollyMmm** [comment]
I also met that status.
$ pod setup
Setting up CocoaPods master repo
[!] /usr/bin/git clone https://github.com/CocoaPods/Specs.git master

Cloning into 'master'...
error: RPC failed; curl 56 SSLRead() return error -9806
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed.

I use many ways to result it ,but they don't work!

1.I use $ pod repo remove master
$ pod repo add master https://gitcafe.com/akuandev/Specs.git
[!] To setup the master specs repo, please run `pod setup`.

I try $pod setup  ,but no answer!
It trapped!

2.I use 

Molly:repos molly$ pod repo remove master
Removing spec repo `master`
Molly:repos molly$ cd ~/.cocoapods/repos
Molly:repos molly$ ls
Molly:repos molly$ ls -al
total 0
drwxr-xr-x  2 molly  staff   68  8 10 21:45 .
drwxr-xr-x  3 molly  staff  102  8 10 17:27 ..
Molly:repos molly$ /usr/bin/git clone git@github.com:CocoaPods/Specs.git master
Cloning into 'master'...
The authenticity of host 'github.com (192.30.253.113)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
Are you sure you want to continue connecting (yes/no)? y
Please type 'yes' or 'no': yes
Warning: Permanently added 'github.com,192.30.253.113' (RSA) to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Molly:repos molly$ ls -al
total 0
drwxr-xr-x  2 molly  staff   68  8 10 21:47 .
drwxr-xr-x  3 molly  staff  102  8 10 17:27 ..
Molly:repos molly$ /usr/bin/git clone https://github.com/CocoaPods/Specs.git master
Cloning into 'master'...
fatal: unable to access 'https://github.com/CocoaPods/Specs.git/': SSLRead() return error -9806
Molly:repos molly$ !] To setup the master specs repo, please run `pod setup`.

 I want your help, thanks!

**Comment by SoftDev0420** [comment]
I installed 1.1.0.rc.2 and did "pod update" on terminal.
It said:
"Update all pods
Re-creating CocoaPods due to major version update.
Updating local specs repositories
Performing a deep fetch of the `master` specs repo to improve future performance"

And it didn't work at all. I tried several time but got same issues.

Does anybody have an idea for this issue?

Thank you.

**Comment by iljaiwas** [comment]
Run into the same issue. Solved by running this command on macOS 10.11:

    sudo gem update -n /usr/local/bin –system

**Comment by zhouzichao0709** [comment]
having the same issue now.
